design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,361 @@
+        billingAddress: NegotiableQuoteBillingAddressInput!
+    ): AssignBillingAddressToNegotiableQuoteOutput @doc(description:"Assigns new billing address to negotiable quote")
+
+    estimateShippingCostForNegotiableQuote(
+        shippingAddress: NegotiableQuoteEstimateShippinAddressInput!
+    ): EstimateShippingCostForNegotiableQuoteOutput @doc(description:"Estimate shipping costs for a negotiable quote")
+
+    estimateShippingCostByAddressId(
+        shippingAddressId: NegotiableQuoteEstimateShippinAddressIdInput!
+    ): EstimateShippingCostByAddressIdOutput @doc(description:"Estimate shipping costs for a negotiable quote by Address ID")
+
+    estimateShippingInformationForNegotiableQuote(
+        addressInformation: NegotiableQuoteShippingInformationInput!
+    ): EstimateShippingInformationForNegotiableQuoteOutput @doc(description:"Get list of payment options and the order totals by address and shipping info")
+
+    setShippingMethodToNegotiableQuote(
+        negotiableQuoteShippingMethod: NegotiableQuoteShippingMethodInput!
+    ): setShippingMethodToNegotiableQuoteOutput @doc(description:"Set negotiable quote shipping method")
+
+    setCouponCodeToNegotiableQuote(
+        negotiableQuoteCouponCode: NegotiableQuoteCouponCodeInput!
+    ): SetCouponCodeToNegotiableQuoteOutput @doc(description:"Set negotiable quote coupon code")
+
+    removeCouponCodeFromNegotiableQuote(
+        cartId: String!
+    ): RemoveCouponCodeFromNegotiableQuoteOutput @doc(description:"Remove negotiable quote coupon code")
+
+    setPaymentInformationForNegotiableQuote(
+        paymentInformation: NegotiableQuotePaymentInformationInput!
+    ): SetPaymentInformationForNegotiableQuoteOutput! @doc(description:"Sets payment information and the billing address for the negotiable quote")
+
+    createOrderWithPaymentInfoNegotiableQuote(
+        paymentInformation: NegotiableQuotePaymentInformationInput!
+    ): CreateOrderWithPaymentInfoNegotiableQuoteOutput! @doc(description:"Assigns payment information and the billing address for the negotiable quote to create order")
+
+    createOrderWithGiftCardForNegotiableQuote(
+        giftCardAccountData: NegotiableQuoteGiftCardAccountDataInput!
+    ): CreateOrderWithGiftCardForNegotiableQuoteOutput! @doc(description:"Assigns GiftCard information and cart ID for the negotiable quote to create order")
+
+    removeGiftCardFromNegotiableQuote(
+        giftCardsData: NegotiableQuotegiftCardsInput!
+    ): RemoveGiftCardFromNegotiableQuoteOutput! @doc(description:"Removes a gift card that has been applied to a negotiable quote")
+}
+
+type Customer {
+        negotiable_quotes(
+        page_size: Int = 20,
+        current_page: Int = 1,
+        filter: NegotiableQuoteFilterInput
+    ): [NegotiableQuote] @doc(description: "Get negotiable quotes of a customer")
+}
+
+type NegotiableQuote {
+    billing_address: BillingAddress! @doc(description: "Billing address of negotiable quote")
```

Comment:
`billing_address` is non-nullable here, but `NegotiableQuoteRequestInput` doesn't take a `BillingAddress`.

What happens if I create a negotiable quote without a `billing_address` and then query for a `NegotiableQuote.billing_address`?

Reply:
Per @mbrinton01 in another comment it should not even be possible to set billing address on the negotiable quote, before proceeding to checkout.